src/common/ScreenExporter.hpp

```diff
@@ -0,0 +1,58 @@
+/* *****************************************************************************
+ * A.L.E (Arcade Learning Environment)
+ * Copyright (c) 2009-2013 by Yavar Naddaf, Joel Veness, Marc G. Bellemare and 
+ *   the Reinforcement Learning and Artificial Intelligence Laboratory
+ * Released under the GNU General Public License; see License.txt for details. 
+ *
+ * Based on: Stella  --  "An Atari 2600 VCS Emulator"
+ * Copyright (c) 1995-2007 by Bradford W. Mott and the Stella team
+ *
+ * *****************************************************************************
+ *  ScreenExporter.hpp 
+ *
+ *  A class for exporting Atari 2600 frames as PNGs.
+ *
+ **************************************************************************** */
+
+#ifndef __SCREEN_EXPORTER_HPP__
+#define __SCREEN_EXPORTER_HPP__ 
+
+#include <string>
+#include "display_screen.h"
+#include "ale_screen.hpp"
+
+class ScreenExporter {
+
+    public:
+
+        /** Creates a new ScreenExporter which can be used to save screens using save(filename). */ 
+        ScreenExporter(ColourPalette &palette);
```

Comment:
There's a bit more flexibility with handing a ScreenExporter to the caller, in the sense that you don't have to record every single frame. For example you might want to record only every k episodes. But maybe that's not needed? 

As Marlos pointed out, ScreenExporter doesn't do everything we might want it to. So it might make sense to restrict its use (e.g. record single frames or all frames), and build a separate visualization suite for more complex operations.